qiskit/synthesis/stabilizer/stabilizer_decompose.py

```diff
@@ -143,7 +143,7 @@ def _calc_pauli_diff_stabilizer(cliff, cliff_target):
     phase.extend(phase_stab)
     phase = np.array(phase, dtype=int)
 
-    A = cliff.symplectic_matrix.astype(int)
+    A = cliff.symplectic_matrix.astype(bool)
```

Comment:
This is already of dtype bool in `Clifford` so we should avoid copying here unless we need to:
```suggestion
    A = cliff.symplectic_matrix.astype(bool, copy=False)
```